File: Prácticas/P4/RUBY/Civitas/lib/casilla_calle.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "casilla.rb"

module Civitas
  class CasillaCalle < Casilla
    attr_reader :titulo
    def initialize(titulo)
      @titulo = titulo
      super(@titulo.get_nombre)
    end
    
    def recibe_jugador(actual,todos)
      if(jugador_correcto(actual,todos) == true)
        informe(actual,todos);
        
        if(@titulo.tiene_propietario == false)
          todos.at(actual).puede_comprar_casilla
        else
          @titulo.tramitar_alquiler(todos.at(actual))
        end
      end
    end
    
    def to_s
      to_s="Titulo #{@titulo}"
    end
  end
end

File: Prácticas/P5/JAVA/Civitas/src/civitas/CasillaCalle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author Jose Padial
 */
public class CasillaCalle extends Casilla{
    private TituloPropiedad titulo;
    
    CasillaCalle(TituloPropiedad titulo){
        super(titulo.getNombre());
        this.titulo = titulo;
    }
    
    TituloPropiedad getTituloPropiedad(){
        return titulo;
    }
    
    @Override
     void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual,todos) == true){
            super.informe(actual,todos);
            if(titulo.tienePropietario() == false){
                todos.get(actual).puedeComprarCasilla();
            }
            else{
                titulo.tramitarAlquiler(todos.get(actual));
            }
        }
    }

    @Override
    public String toString() {
        return "CasillaCalle{" + "titulo=" + titulo + '}';
    }
}
